### ISSUE TITLE
AttributeError: 'SpatialReference' object has no attribute 'this'

### ISSUE THREAD

**Issue by DRjy** [OSGeo/gdal#2893]
##   name '__builtin__' is not defined

For example (please modify !!!): I expected to create a object by using the code> `srs=osr. SpatialReference()`, some errors occured.

## Steps to reproduce the problem.

For example (please modify !!!): 

```
>from osgeo import osr
srs = osr.SpatialReference()
```

After execute the two lines of codes, the errors are as following:

```
AttributeError                            Traceback (most recent call last)
D:\Programs\GDAL3\bin\gdal\python\osgeo\osr.py in __init__(self, *args, **kwargs)
    932         try:
--> 933             self.this.append(this)
    934         except __builtin__.Exception:

AttributeError: 'SpatialReference' object has no attribute 'this'
```

During handling of the above exception, another exception occurred:

```
NameError                                 Traceback (most recent call last)
<ipython-input-3-ca735fe97fa3> in <module>
      1 from osgeo import gdal,gdalnumeric,ogr,osr,gdalconst
----> 2 srs = osr.SpatialReference()
      3 # srs.ImportFromEPSG(4326)

D:\Programs\GDAL3\bin\gdal\python\osgeo\osr.py in __init__(self, *args, **kwargs)
    932         try:
    933             self.this.append(this)
--> 934         except __builtin__.Exception:
    935             self.this = this
    936 

NameError: name '__builtin__' is not defined
```

## Operating system

For example (please modify !!!): Windows 10 64 bit

## GDAL version and provenance

For example (please modify !!!): the 3.1.2

**Comment by jratike80** [comment]
Please do not create bug reports if you need help. Use gdal-dev mailing list or gis.stackexchange instead.

I can run those two lines without errors on my OSGeo4W installation. My Python is version 2.7.14 and GDAL is 3.0.3. There seems to be something wrong with your installation. How did you install?

**Comment by DRjy** [comment]
I create a virtual environment named py36 by using `conda` command . I run the command line`conda install -c conda-forge gdal`. I installed the stable releases named `release-1911-x64-gdal-3-1-2-mapserver-7-6-1` from the web site:`http://www.gisinternals.com/release.php`, and create the corresponding envrionment variables.

**Comment by jratike80** [comment]
Because you seem to use Windows perhaps you could install OSGeo4W as a reference and check how it is setting the environment variables and paths. If that does not help write mail to gdal-dev or ask a question in gis.stackexchage "How to install gdal with Python bindings on Conda?". Those forums have much more readeds and for example I know hardly anything about conda.

The importance of using user forums for asking questions is not only that more people can read your question and help you, but also that more people can find how your problem was finally resolved and help themselves with similar problem. GitHub issues are not top hits for the search engines.

**Comment by rouault** [comment]
I suspect something specific to your installation. Mixing GDAL from conda-forge and from gisinternals isn't going to work very well

**Comment by py-qgis-user** [comment]
I encountered a similar problem as OP with V3.1.2 and went to stackexchange with it. I found a solution to the problem, but am not sure whether or not I might have messed up other things. Could somebody verify the solution in [this post](https://gis.stackexchange.com/questions/373309/gdal3-gdal2tiles-attributeerror-spatialreference-object-has-no-attribute-thi/373326#373326)?

Might it be a wrong version of osr.py is being shipped in some packages?
